pkg/middleware/auth_test.go

```diff
@@ -32,6 +34,46 @@ func TestMiddlewareAuth(t *testing.T) {
 			})
 		})
 
+		Convey("Anonymous auth enabled", func() {
+			setting.AnonymousEnabled = true
+			setting.AnonymousOrgName = "test"
```

Comment:
Global state changes should be cleaned up.
```suggestion
			origEnabled := setting.AnonymousEnabled
			t.Cleanup(func() {
				setting.AnonymousEnabled = origEnabled
			})
			origName := setting.AnonymousOrgName
			t.Cleanup(func() {
				setting.AnonymousOrgName = origName
			})
			setting.AnonymousEnabled = true
			setting.AnonymousOrgName = "test"
```